src/Compilers/VisualBasic/Portable/Symbols/Source/SourceParameterSymbol.vb

```diff
@@ -264,6 +264,31 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             Return MyBase.EarlyDecodeWellKnownAttribute(arguments)
         End Function
 
+        Friend Overrides Iterator Function GetCustomAttributesToEmit(compilationState As ModuleCompilationState) As IEnumerable(Of VisualBasicAttributeData)
```

Comment:
Has the scenario where `CallerArgumentExpressionAttribute` is applied multiple times been considered (this requires to define a non-standard attribute)? The below code seems to assume only one usage.
As an example, for `AsyncMethodBuilderAttribute` only the first application is consisdered (the others are ignored).
This method and `CallerArgumentExpressionParameterIndex` probably need to agree on behavior (and not crash ;-).

Such situation could be combined with an application of the attribute using a different constructor (breaks assertion below).

Reply:
> Has the scenario where `CallerArgumentExpressionAttribute` is applied multiple times been considered (this requires to define a non-standard attribute)? The below code seems to assume only one usage.
> As an example, for `AsyncMethodBuilderAttribute` only the first application is consisdered (the others are ignored).
> This method and `CallerArgumentExpressionParameterIndex` probably need to agree on behavior (and not crash ;-).

Applying multiple times is interesting. I'll add a test. Not sure if this scenario will need a warning?

> Such situation could be combined with an application of the attribute using a different constructor (breaks assertion below).

I assumed this can't happen because the attribute description defines what constructor signature should it be:

https://github.com/dotnet/roslyn/blob/fd2a03f8b035b1a6c409f2fdca2a5cf13d7cf356/src/Compilers/Core/Portable/Symbols/Attributes/AttributeDescription.cs

But definitely this worth a test.